### DISCUSSION TITLE
ERROR:  permission denied for schema pg_catalog

### DISCUSSION THREAD

**Discussion by alpaslanakdag** [cybertec-postgresql/pgwatch2#794] [Q&A]
Hi all

I am trying to make a custom installation but when i run below code  i get an error with permission denied.
`psql -f roll_out_metric_time.psql pgwatch2_metrics`

Here is the error.
`psql:00_schema_base.sql:63: ERROR:  permission denied for schema pg_catalog`

Did i miss something? 
Thanks in advance

OS: RHEL 9.3
postgres v 13.14
pgwatch2 v1.13.0

**Comment by pashagolub** [comment]
Helo.

`01_old_metrics_cleanup_procedure.sql` uses functions from `pg_catalog` which **must** be available under normal conditions. Make sure your user have proper permissions.

**Reply by alpaslanakdag** [reply]
thank you for your answer.
I run this command with postgres user and it is superuser. Which user should it be?

**Reply by pashagolub** [reply]
Something really strange happens in your environment. 

It's better to run script under dedicated `pgwatch2` user.
https://pgwatch2.readthedocs.io/en/latest/custom_installation.html#config-db-based-setup

**Reply by alpaslanakdag** [reply]
I have already try it and i get the same error, plus a warning.
`psql:00_schema_base.sql:9: WARNING:  no privileges were granted for "public"`

**Reply by pashagolub** [reply]
`pgwatch2` must be the owner of the metric database.

**Reply by alpaslanakdag** [reply]
It is the owner of pgwatch2_metrics database
